src/Adapter/Profile/Permission/QueryHandler/GetPermissionsForConfigurationHandler.php

```diff
@@ -0,0 +1,356 @@
+            Permission::VIEW,
+            Permission::ADD,
+            Permission::EDIT,
+            Permission::DELETE,
+        ];
+
+        $tabPermissionsForProfiles = $this->getTabPermissionsForProfiles($profiles);
+        $modulePermissionsForProfiles = $this->getModulePermissionsForProfiles($profiles);
+
+        $permissionIds = array_flip(Permission::SUPPORTED_PERMISSIONS);
+        $employeeProfileId = $query->getEmployeeProfileId()->getValue();
+        $canEmployeeEditPermissions = $this->authorizationChecker->isGranted(PageVoter::UPDATE, 'AdminAccess');
+
+        $bulkConfigurationPermissions = $this->getBulkConfigurationForProfiles(
+            $employeeProfileId,
+            $canEmployeeEditPermissions,
+            $tabPermissionsForProfiles,
+            $profiles,
+            $tabs,
+            $permissions
+        );
+
+        return new ConfigurablePermissions(
+            $tabPermissionsForProfiles,
+            $modulePermissionsForProfiles,
+            $profiles,
+            $tabs,
+            $bulkConfigurationPermissions,
+            $permissions,
+            $permissionIds,
+            $employeeProfileId,
+            $canEmployeeEditPermissions
+        );
+    }
+
+    /**
+     * @return array<int> IDs of non configurable tabs
+     */
+    private function getNonConfigurableTabs(): array
+    {
+        return [
+            Tab::getIdFromClassName('AdminLogin'),
+            Tab::getIdFromClassName('AdminQuickAccesses'),
+            Tab::getIdFromClassName('AdminTabs'),
+        ];
+    }
+
+    /**
+     * @return array<array<string, int|string|bool>>
+     */
+    private function getProfilesForPermissionsConfiguration(): array
+    {
+        $legacyProfiles = Profile::getProfiles($this->languageId);
+        $profiles = [];
+
+        foreach ($legacyProfiles as $profile) {
+            $profiles[] = [
+                'id' => $profile['id_profile'],
+                'name' => $profile['name'],
+                'is_administrator' => (int) $profile['id_profile'] === _PS_ADMIN_PROFILE_,
+            ];
+        }
+
+        return $profiles;
+    }
+
+    /**
+     * @return array
+     */
+    private function getTabsForPermissionsConfiguration(): array
+    {
+        $nonConfigurableTabs = $this->getNonConfigurableTabs();
+        $legacyTabs = Tab::getTabs($this->languageId);
+        $tabs = [];
+
+        foreach ($legacyTabs as $tab) {
+            // Don't allow permissions for unnamed tabs (ie. AdminLogin)
+            if (empty($tab['name'])) {
+                continue;
+            }
+
+            // Don't allow permissions for undefined parents
+            if ($tab['id_parent'] === '-1') {
+                continue;
+            }
+
+            if (in_array($tab['id_tab'], $nonConfigurableTabs)) {
+                continue;
+            }
+
+            $this->whitelist[] = $tab['id_tab'];
+            $tabs[] = [
+                'id' => $tab['id_tab'],
```

Comment:
```suggestion
                'id' => (int) $tab['id_tab'],
```

Maybe we should force casting here? Especially since it comes from a legacy function that returns the SQL query as is, changes are the values will be string no?